File: src/main/java/dev/lampirg/consultationappointment/data/teacher/TeacherRepository.java
package dev.lampirg.consultationappointment.data.teacher;

import dev.lampirg.consultationappointment.data.model.PersonRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface TeacherRepository extends PersonRepository<Teacher> {
    @Query("select t from DatePeriod t where t.id = ?1")
    Optional<DatePeriod> findDatePeriodById(Long datePeriodId);
}

File: src/main/kotlin/dev/lampirg/consultationappointment/data/teacher/TeacherRepository.kt
package dev.lampirg.consultationappointment.data.teacher

import dev.lampirg.consultationappointment.data.model.PersonRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface TeacherRepository : PersonRepository<Teacher> {
    @Query("select t from DatePeriod t where t.id = ?1")
    fun findDatePeriodById(datePeriodId: Long): Optional<DatePeriod>
}